topdown/crypto.go

```diff
@@ -390,10 +423,10 @@ func getRSAPrivateKeyFromString(key string) (interface{}, error) {
 		return nil, err
 	}
 
-	return getRSAPrivateKeyFromPEM(b64)
+	return getPrivateKeyFromPem(b64)
 }
 
-func getRSAPrivateKeyFromPEM(pemBlocks []byte) (interface{}, error) {
+func getPrivateKeyFromPem(pemBlocks []byte) (interface{}, error) {
 
 	// decode the pem into the Block struct
 	p, _ := pem.Decode(pemBlocks)
```

Comment:
suggestion: I know this isn't really your change, but while we're here... 

I think we can improve `failed to parse PEM block containing the key`. I'd suggest: `No PEM data found in input`.

⚠️ Scope Creep ™️ ⚠️ If you are feeling adventurous, you might consider altering this function to support the input of multiple PEM blocks and returning the first one that's a private key. But this is 100% not needed for this PR 😆

Reply:
sure! but given a list of many PEMs that happens to be a private key, we should append a list and return these, not just the first one. I'll make a commit for it! :)

Reply:
see ed061d911d197010d963cf7e0cecbb30000e1c36

Reply:
This does raise an interesting question: should we accept a list of private keys as we do for certificates in `crypto.x509.parse_certificates`?

I'd have done it more like this: https://go.dev/play/p/2XqZn1wbgC2, this way handles the input containing more than one PEM block.

I'm not sure what we should do when there are A) other PEM block types, and B) More than one key.

I think for A) we can just roll over them and ignore

For B), perhaps it does make sense to have this function handle the parsing of many private keys at once. This way we handle the multiple and the single case pretty easily.

```
-----BEGIN PRIVATE KEY-----
 MC4CAQAwBQYDK2VwBCIEIJHG93jlLLLTF6Stky5+8Q7mMpgCkYYTO12NDAzlJn3w
-----END PRIVATE KEY-----
-----BEGIN PRIVATE KEY-----
 MC4CAQAwBQYDK2VwBCIEIJHG93jlLLLTF6Stky5+8Q7mMpgCkYYTO12NDAzlJn3w
-----END PRIVATE KEY-----
...
```

or the simple case:

```
-----BEGIN PRIVATE KEY-----
 MC4CAQAwBQYDK2VwBCIEIJHG93jlLLLTF6Stky5+8Q7mMpgCkYYTO12NDAzlJn3w
-----END PRIVATE KEY-----
```

Let me know what you think.

Reply:
I think I'm happy with the solution that I proposed in https://github.com/open-policy-agent/opa/commit/ed061d911d197010d963cf7e0cecbb30000e1c36,  it will ignore other PEM types, see test(https://github.com/open-policy-agent/opa/pull/6084/commits/ed061d911d197010d963cf7e0cecbb30000e1c36#diff-d47866b4e7252abbde5b22fe032a80210ec24dbbb0572ede2a12d4eaeda7a683R540) and return the valid keys if there are more than one...

Reply:
I guess what feels unexpected to me is that `getPrivateKeyFromPem` accepts `[][]byte` but is only ever used with a single `[]byte`.

I think that it might also break `builtinCryptoX509ParseRSAPrivateKey` (soon to be renamed `builtinCryptoJWKFromPrivateKey`) as that caller expects a key rather than a slice of keys.

Reply:
I will make a PR for you later today 👍

Reply:
👍 thanks!

Reply:
(it might be tomorrow morning my time (UK) before I get back to you)

Reply:
I think that'll do it!